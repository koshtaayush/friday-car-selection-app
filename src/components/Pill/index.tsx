import React from 'react';
import styled from 'styled-components';

interface Props {
    value: string
    onClickProp: (element: string) => void
}

const Pill: React.FC<Props> = (props) => {

    const { value, onClickProp } = props
    return (
        <React.Fragment>
            <PillContainer onClick={() => onClickProp(value)}>
                <Name>{value.toLowerCase()}</Name>
            </PillContainer>
        </React.Fragment>
    )
}

export default Pill

const PillContainer = styled.div`
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0.5rem;
    width: 30%;
    border: 1px solid #e1e1e1;
    margin: 0.25rem;
    border-radius: 4px;

    @media only screen and (max-width: 600px) {
        width: 100%;
    }

    &:hover {
        border: 1px solid #468FF7;
        background: #468FF7;
        color: #fff;

    }
`

const Name = styled.div`
    height: 10%;
    width: 100%;
    text-transform: capitalize;
    display: flex;
    align-items: center;
    justify-content: center;
`