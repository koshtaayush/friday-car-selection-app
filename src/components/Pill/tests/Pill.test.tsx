import React from 'react'
import { shallow } from 'enzyme'

import Pill from '..'

describe('Pill', () => {
    it('Should match the snapshot', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Pill
                value={'Car Name'}
                onClickProp={mockFn}
            />
        )

        expect(component).toMatchSnapshot()
    })

    it('Should render the Pill text correctly', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Pill
            value={'Car Name'}
            onClickProp={mockFn}
        />
        )

        expect(component.find({ 'test-id': 'name' }).text()).toEqual('car name')
    })
    
    it('Should execute click method on click event correctly', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Pill
            value={'Car Name'}
            onClickProp={mockFn}
        />)

        component.find({ 'test-id': 'pillContainer'}).simulate("click")
        expect(mockFn).toHaveBeenCalled()
    })

    
})
