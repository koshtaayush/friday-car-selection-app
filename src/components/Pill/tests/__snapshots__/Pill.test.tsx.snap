// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill Should match the snapshot 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
    test-id="pillContainer"
  >
    <styled.div
      test-id="name"
    >
      car name
    </styled.div>
  </styled.div>
</Fragment>
`;
